Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -441,10 +443,10 @@ public class PBXProjGenerator {
                     )
                     targetFrameworkBuildFiles.append(buildFile.reference)
                 }
-
-                if (dependency.embed ?? target.type.isApp) && !dependencyTarget.type.isLibrary {
-
-
+                
+                let shouldEmbed = target.type.isApp
+                    || (target.type.isTest && (dependencyTarget.type.isFramework || dependencyTarget.type == .bundle))
+                if (dependency.embed ?? shouldEmbed) && !dependencyTarget.type.isLibrary {
```

Comment:
This makes it so you don't have to explicitly list embed: true for test targets when it's a framework or bundle. It already does this for Carthage.